ui/js/controllers/perf/alerts.js

```diff
@@ -0,0 +1,149 @@
+"use strict";
+
+perf.factory('PhAlerts', ['$http', 'thServiceDomain', function($http, thServiceDomain) {
+    return {
+        getAlertSummary: function(id) {
+            return $http.get(thServiceDomain +
+                             '/api/performance/alertsummary?id=' + id).then(
+                                 function(response) {
+                                     return response.data;
+                                 });
+        },
+        getAlertSummaries: function(offset, count) {
+            return $http.get(thServiceDomain +
+                             '/api/performance/alertsummary?offset=' + offset +
+                             '&count=' + count).then(function(response) {
+                                 return response.data;
+                             });
+        }
+    };
+}]);
+
+perf.controller('AlertsCtrl', [
+    '$state', '$stateParams', '$scope', '$rootScope', '$http', '$q', 'thUrl',
+    'ThRepositoryModel', 'ThOptionCollectionModel', 'ThResultSetModel',
+    'thDefaultRepo', 'PhSeries', 'PhAlerts', 'phTimeRanges', 'dateFilter', 'thDateFormat',
+    function AlertsCtrl($state, $stateParams, $scope, $rootScope, $http, $q,
+                        thUrl, ThRepositoryModel, ThOptionCollectionModel,
+                        ThResultSetModel, thDefaultRepo, PhSeries, PhAlerts,
+                        phTimeRanges, dateFilter, thDateFormat) {
+
+        $scope.alertSummaries = [];
+        $scope.getCappedMagnitude = function(percent) {
+            // arbitrary scale from 0-20% multiplied by 5, capped
+            // at 100 (so 20% regression == 100% bad)
+            return Math.min(Math.abs(percent)*5, 100);
+        };
+
+        function addAlertSummaries(alertSummaries) {
+            var resultSetToSummaryMap = {};
+
+            alertSummaries.forEach(function(alertSummary) {
+                if (!resultSetToSummaryMap[alertSummary.repository]) {
+                    resultSetToSummaryMap[alertSummary.repository] = {};
+                }
+                _.forEach([alertSummary.result_set_id, alertSummary.prev_result_set_id],
+                          function(resultSetId) {
+                              if (!resultSetToSummaryMap[alertSummary.repository][
+                                  resultSetId]) {
```

Comment:
I know you are keeping this in the <80 character limits, but it is hard to read with a line break after the [